pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
```

Comment:
Please add a nix-shell shebang that uses python from within *this* nixpkgs revision the script is in. We do not want some impurity causing subtile difference in JSON formatting, sorting, filtering, ...

Reply:
Is 

```sh
#!/usr/bin/env nix-shell
#!nix-shell -i python3 -p python3
```

enough? If not, please point me to some examples of what you mean.

Reply:
Maybe add a line with
```
#!nix-shell -I nixpkgs=../../../..
```
directly below what you wrote. I think that works although it looks super ugly. (I have this from the nix-shell man page.)